CombinatorialGames/Surreal/Real.lean

```diff
@@ -195,12 +247,19 @@ theorem toIGame_add_ratCast_equiv (x : ℝ) (q : ℚ) : toIGame (x + q) ≈ x +
     · rw [← IGame.lt_sub_iff_add_lt, ← (IGame.ratCast_sub_equiv ..).lt_congr_right]
       simpa [lt_sub_iff_add_lt]
     · simpa
-  · rw [← IGame.lt_sub_iff_add_lt, ← (IGame.ratCast_sub_equiv ..).lt_congr_right]
+  · rw [r.toIGame_equiv_ratCast.lt_congr_right, ← IGame.lt_sub_iff_add_lt,
+      ← (IGame.ratCast_sub_equiv ..).lt_congr_right]
     simpa [lt_sub_iff_add_lt]
 
 theorem toIGame_ratCast_add_equiv (q : ℚ) (x : ℝ) : toIGame (q + x) ≈ q + x := by
   simpa [add_comm] using toIGame_add_ratCast_equiv x q
 
+theorem toIGame_add_dyadic_equiv (x : ℝ) (q : Dyadic) : toIGame (x + q) ≈ x + q :=
+  (toIGame_add_ratCast_equiv _ _).trans (add_congr_right q.toIGame_equiv_ratCast.symm)
+
+theorem toIGame_dyadic_add_equiv (q : Dyadic) (x : ℝ) : toIGame (q + x) ≈ q + x := by
+  simpa [add_comm] using toIGame_add_dyadic_equiv x q
+
 theorem toIGame_add_equiv (x y : ℝ) : toIGame (x + y) ≈ x + y := by
   rw [AntisymmRel, le_iff_forall_lf, le_iff_forall_lf, forall_leftMoves_add, forall_rightMoves_add]
   simp_rw [forall_leftMoves_toIGame, forall_rightMoves_toIGame, Numeric.not_le]
```

Comment:
(`forall_leftMoves_toIGame` and `forall_rightMoves_toIGame` can be moved to `rw`. However, this is a little less nice - is that why we avoid it here?)

Reply:
That was the idea, yeah. We can trim one line off the code by doing this.